InnosentStateMachine.js

```diff
@@ -0,0 +1,128 @@
+
+      switch (states.state) {
+        case STATE.RESET:
+          // Waits for the door to close before restarting the state machine
+          // eslint-disable-next-line eqeqeq
+          if (door.signal == DOOR_STATE.CLOSED) {
+            state = STATE.NO_PRESENCE_NO_SESSION
+          }
+          break
+        case STATE.NO_PRESENCE_NO_SESSION:
+          // Checks the average XeThru Value over the last 15 seconds
+
+          // eslint-disable-next-line eqeqeq
+          if (door.signal == DOOR_STATE.OPEN) {
+            state = STATE.DOOR_OPENED_START
+          }
+
+          // Waits for both the XeThru and motion sensor to be active
+          // Removing the condition for motion sensor to be active to go into this loop since we don't have a motion sensor in this space
+          else if (inPhase_avg > mov_threshold && doorDelay > DOOR_THRESHOLD_MILLIS) {
+            state = STATE.MOTION_DETECTED
+          }
+          break
+        case STATE.DOOR_OPENED_START:
+          // eslint-disable-next-line eqeqeq
+          if (door.signal == DOOR_STATE.CLOSED) {
+            if (doorDelay < DOOR_THRESHOLD_MILLIS) {
+              state = STATE.DOOR_OPENED_START
+            }
+            if (doorDelay >= DOOR_THRESHOLD_MILLIS && inPhase_avg > mov_threshold) {
+              state = STATE.DOOR_CLOSED_START
+            } else {
+              state = STATE.NO_PRESENCE_NO_SESSION
+            }
+          }
+          break
+        case STATE.DOOR_CLOSED_START:
+          if (inPhase_avg > mov_threshold) {
+            state = STATE.MOVEMENT
+          }
+          break
+        case STATE.MOTION_DETECTED:
+          state = STATE.MOVEMENT
+          break
+        case STATE.DOOR_OPENED_CLOSE:
+          state = STATE.RESET
+          break
+        case STATE.MOVEMENT:
+          // eslint-disable-next-line eqeqeq
+          if (door.signal == DOOR_STATE.OPEN && session.od_flag == OD_FLAG_STATE.NO_OVERDOSE) {
+            state = STATE.DOOR_OPENED_CLOSE
+          } else if (inPhase_avg < mov_threshold) {
+            state = STATE.STILL
+          }
+          // eslint-disable-next-line eqeqeq
+          if (session.od_flag == OD_FLAG_STATE.NO_OVERDOSE && (await db.isOverdoseSuspectedInnosent(session, location_data))) {
+            state = STATE.SUSPECTED_OD
+          }
+
+          break
+        case STATE.STILL:
+          // eslint-disable-next-line eqeqeq
+          if (door.signal == DOOR_STATE.OPEN && session.od_flag == OD_FLAG_STATE.NO_OVERDOSE) {
+            state = STATE.DOOR_OPENED_CLOSE
+          } else if (inPhase_avg > mov_threshold) {
+            state = STATE.MOVEMENT
+          }
+          // eslint-disable-next-line eqeqeq
+          if (session.od_flag == OD_FLAG_STATE.NO_OVERDOSE && (await db.isOverdoseSuspectedInnosent(session, location_data))) {
+            state = STATE.SUSPECTED_OD
+          }
+
+          break
```

Comment:
I think that these two `case` statements could be made into a single one:

```
case STATE.MOVEMENT:
case STATE.STILL:
  if (session.od_flag...) {
     state = STATE.SUSPECTED_OD
  } else if (door.signal...) {
     state = STATE.DOOR_OPENED_CLOSE
  } else if (inPhase_avg < mov_threshold) {
     state = STATE.STILL
  } else if (inPhase_avg > mov_threshold) {
    state = STATE.MOVEMENT
  }
  break;
```

Also, should one of `else if (inPhase_avg < mov_threshold) ` or `(inPhase_avg > mov_threshold)` be `<=` or `>=`?